### PR TITLE
[v6r18] Fix Waiting site name and RMS requests finalization

### PR DESCRIPTION
**Pull request by phicharp**
* Attempt to get more precise site name while the job is Waiting, using the set of sites at which the input files are online rather than checking Tier1s in eligible sites

* Allow to finalize a request form a script, setting status of the job to Done if the request is Done

* Change the bucket sizes for the monitoring plots as function of the time span. To Do: allow this to be defined in the CS, if it is felt necessary...

### CODE REVIEW

**Comment by atsareg** [comment]
I do not think that dirac-rms-show-request can update the request or any other data. It is better to introduce a new command, e.g. dirac-rms-edit(update,modify)-request, which allows to modify requests and/or associated jobs. BTW, modification of requests is a demanded feature, as far as I remember

**Comment by atsareg** [comment]
The Waiting Site logic becomes more and more involved and should be documented appropriately.

**Comment by phicharp** [comment]
I agree with you about diracyrms-show-request, but it was already modifying the requests eventually.. The reason is that 95% of the code to parse, display etc... is the same and it was much easier to add the 2 necessary lines to an existing script than moving the common code out for sharing and split the scripts... I will try and do that though (note that there is a dirac-rms-reset-request that "duplicates" the --Reset option of the show-request script... Needs to be cleaned up and make more rational...

I'll try and document the whole story about site selection etc... My main problem with documentation is 2 fold:
1. where to put it. Asking 4 persons you get 4 answers
2. formatting sphynx documentation is just pure %$@!!@$! but I repeat myself here ;-)

File: ResourceStatusSystem/Command/SpaceTokenOccupancyCommand.py [hunk 1]
```diff
@@ -100,8 +100,8 @@ def doNew( self, masterParams = None ):
     # 10 secs of timeout. If it works, the reply is immediate.
     occupancyResult = pythonCall( 10, lcg_util.lcg_stmd, spaceToken, spaceTokenEndpoint, True, 0 )
     if not occupancyResult[ 'OK' ]:
-      self.log.error("Could not get spaceToken occupancy", "Endpoint/spaceToken: %s/%s" %(spaceTokenEndpoint, spaceToken),
-                     occupancyResult['Message'])
+      self.log.error( "Could not get spaceToken occupancy from endPoint/spaceToken: %s/%s" % ( spaceTokenEndpoint, spaceToken ),
```
Comment (fstagni) [inline ResourceStatusSystem/Command/SpaceTokenOccupancyCommand.py]: No, please do:
```
      self.log.error("Could not get spaceToken occupancy",
                     "Endpoint/spaceToken: %s/%s ==> %s" %(spaceTokenEndpoint, spaceToken, occupancyResult['Message']))
```

Comment (phicharp) [inline ResourceStatusSystem/Command/SpaceTokenOccupancyCommand.py]: Done

**Comment by phicharp** [comment]
@atsareg: I looked at how to make a dirac-rms-fix-request but this would become absurd, since it would just be a copy of the current dirac-rms-show-request and dirac-rms-show-request would just be the same removing the options to fix the request (--Reset and --FixJob)
I can just rename show-request and call it -fix-request (or -modify-request), knowing that if no option is set to fix the request it will display it... Having both doesn't make much sense, does it?

**Comment by atsareg** [comment]
May be we can then rename it to just dirac-rms-request and make it a kind of swiss army knife tool for showing, resetting and editing depending on switches.

**Comment by phicharp** [comment]
If needed I can also move it to v6r18 (still integration) as this is not urgent. Just let me know...

**Comment by chaen** [comment]
I was just about to propose that, since the script name changes

**Comment by atsareg** [comment]
May be then dirac-rms-reset-request can be also merged into dirac-rms-request ? It also has much in common. Then there can be options --show, --reset, --update as subcommands, for example. This is a frequent pattern now, e.g. git

**Comment by phicharp** [comment]
I think the whole functionality of reset-request is available in the other generic script, therefore it can probably just be removed... I'll check thoroughly though

**Comment by atsareg** [comment]
Also then check the dirac-rms-cancel-request command please

**Comment by phicharp** [comment]
Merger done and useless scripts removed
dirac-rms-reset-request replaced by dirac-rms-request --Reset
dirac-rms-cancel-request replaced by dirac-rms-request --Cancel

**Comment by fstagni** [comment]
Looks OK to me and I agree it should be moved to v6r18.

**Pull request merged**